pallets/dca/src/lib.rs

```diff
@@ -943,14 +950,36 @@ impl<T: Config> Pallet<T> {
 		let fee_currency = schedule.order.get_asset_in();
 		let fee_amount_in_sold_asset = Self::convert_weight_to_fee(weight_to_charge, fee_currency)?;
 
-		Self::unallocate_amount(schedule_id, schedule, fee_amount_in_sold_asset)?;
+		if T::SwappablePaymentAssetSupport::is_transaction_fee_currency(fee_currency) {
+			Self::unallocate_amount(schedule_id, schedule, fee_amount_in_sold_asset)?;
 
-		T::Currencies::transfer(
-			fee_currency,
-			&schedule.owner,
-			&T::FeeReceiver::get(),
-			fee_amount_in_sold_asset,
-		)?;
+			T::Currencies::transfer(
+				fee_currency,
+				&schedule.owner,
+				&T::FeeReceiver::get(),
+				fee_amount_in_sold_asset,
+			)?;
+		} else {
+			//We buy DOT with insufficient asset, for the treasury
+			//The DOT we need to buy is calculated the same way how we convert weight to insufficient fee
+			let pool_trade_fee = T::SwappablePaymentAssetSupport::calculate_fee_amount(fee_amount_in_sold_asset)?;
+
+			//Since there is a trade fee involved in xyk buy swap, we need to unallocate that, together with amount_in
+			let effective_amount_in = fee_amount_in_sold_asset
+				.checked_add(pool_trade_fee)
+				.ok_or(ArithmeticError::Overflow)?;
+			Self::unallocate_amount(schedule_id, schedule, effective_amount_in)?;
+
+			let fee_in_dot = Self::convert_to_polkadot_native_asset(Self::weight_to_fee(weight_to_charge))?;
```

Comment:
do you need to do this again ? 

should not this be equal to fee_amount_in_sold_asset?

Reply:
`fee_amount_in_sold_asset` is different from `fee_in_dot`.

Because the `amount_in` (so DCA fee asset) is not necessarly DOT.

But we need to calculate the `fee_in_dot`, so we know how much we should do a buy trade for, within the FeeAsset-DOT xyk pool